pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -368,15 +368,19 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListenerForPortOrUDS(li
 	listenerOpts.class = ListenerClassSidecarInbound
 
 	if old, exists := listenerMap[listenerOpts.port.Port]; exists {
-		// If we already setup this hostname, its not a conflict. This may just mean there are multiple
-		// IPs for this hostname
-		if old.instanceHostname != pluginParams.ServiceInstance.Service.Hostname {
+		if old.protocol != listenerOpts.port.Protocol && old.instanceHostname != pluginParams.ServiceInstance.Service.Hostname {
```

Comment:
are protocol and hostname the *only* fields we use to construct the listener? if it is, can we just pass those in instead of all this other info which is confusing and may lead to bugs? If its not, isn't this not safe still since we could have a protocol and hostname match but something else mismatch?

Reply:
I think it could not occur different protocols for k8s service, udp has been filtered out.

Reply:
Sorry. I did not understand. Are you saying protocol check is not needed?
Also I could not think of any thing else that could cause conflict

```
	push              *model.PushContext
	proxy             *model.Proxy
	bind              string
	port              *model.Port
	filterChainOpts   []*filterChainOpts
	bindToPort        bool
	skipUserFilters   bool
	needHTTPInspector bool
	class             ListenerClass
	service           *model.Service
	protocol          istionetworking.ListenerProtocol
```
Here is what it is based  on?  @howardjohn @hzxuzhonghu any thing else we should care about for conflicts?